pkg/schedule/schedule.go

```diff
@@ -162,6 +172,59 @@ func Init() error {
 	if err != nil {
 		return err
 	}
+
+	testMode := os.Getenv(StorkTestModeEnvVariable)
+	if testMode == "true" {
+
+		fn := func(object runtime.Object) error {
+			cm, ok := object.(*v1.ConfigMap)
+			if !ok {
+				err := fmt.Errorf("invalid object type on configmap watch: %v", object)
+				return err
+			}
+
+			if len(cm.Data) > 0 {
+				timeString := cm.Data[MockTimeConfigMapKey]
+				if len(timeString) > 0 {
+					t, err := time.Parse(time.RFC1123, timeString)
+					if err != nil {
+						err = fmt.Errorf("failed to parse time in mock config map due to: %v", err)
+						logrus.Errorf(err.Error())
+						return err
+					}
+
+					logrus.Infof("Setting mock time to: %v current time: %v", t, GetCurrentTime())
+					setMockTime(&t)
+				} else {
+					logrus.Infof("time string is empty. Resetting mock time")
+					setMockTime(nil)
+				}
+			}
+
+			return nil
+		}
+
+		logrus.Infof("Stork test mode enabled. Watching for config map: %s for mock times", MockTimeConfigMapName)
+		cm, err := k8s.Instance().GetConfigMap(MockTimeConfigMapName, MockTimeConfigMapNamespace)
+		if err != nil {
+			if errors.IsNotFound(err) {
```

Comment:
We should not depend on it being created before stork is started. If the parameter is passed in that it is running in test mode, go ahead and watch the config map always. Create the config map if it doesn't exist.